src/Symfony/Component/Validator/ConstraintValidatorInterface.php

```diff
@@ -30,6 +31,8 @@ public function initialize(ExecutionContextInterface $context);
      *
      * @param mixed      $value      The value that should be validated
      * @param Constraint $constraint The constraint for the validation
+     *
+     * @throws ValidatorException
```

Comment:
yet some implementations are tied to the constraint or metadata factory even..

- NoSuchMetadataException
- MissingOptionsException

I'd prefer the expected ones here:

- UnexpectedTypeException
- UnexpectedValueException
- ConstraintDefinitionException

Reply:
> We cant just add ANY exception

What I have added here is not ANY exception.. It is a specif exception that ANY exception implemented by a developer it must extend it...

Reply:
ok put different :) i think `ConstraintValidatorInterface::validate()` is designed to throw specific exceptions, for specific cases. Not just any `ValidatorException`.. esp. since the high level validator checks specific ones, e.g. `UnexpectedValueException`

Throwing ValidatorException instead of UnexpectedValueException would be a wrong implementation. As such i think we should document the specific cases.

Reply:
Now the argument is understandable.. although I disagree :)